api/src/main/java/org/apache/iceberg/expressions/Expressions.java

```diff
@@ -74,37 +73,32 @@ public static Expression not(Expression child) {
 
   @SuppressWarnings("unchecked")
   public static <T> UnboundTerm<T> bucket(String name, int numBuckets) {
-    Transform<?, T> transform =
-        (Transform<?, T>) Transforms.bucket(Types.StringType.get(), numBuckets);
+    Transform<?, T> transform = (Transform<?, T>) Transforms.bucket(numBuckets);
     return new UnboundTransform<>(ref(name), transform);
   }
 
   @SuppressWarnings("unchecked")
   public static <T> UnboundTerm<T> year(String name) {
-    return new UnboundTransform<>(
-        ref(name), (Transform<?, T>) Transforms.year(Types.TimestampType.withZone()));
+    return new UnboundTransform<>(ref(name), (Transform<?, T>) Transforms.year());
   }
 
   @SuppressWarnings("unchecked")
   public static <T> UnboundTerm<T> month(String name) {
-    return new UnboundTransform<>(
-        ref(name), (Transform<?, T>) Transforms.month(Types.TimestampType.withZone()));
+    return new UnboundTransform<>(ref(name), (Transform<?, T>) Transforms.month());
   }
 
   @SuppressWarnings("unchecked")
   public static <T> UnboundTerm<T> day(String name) {
-    return new UnboundTransform<>(
-        ref(name), (Transform<?, T>) Transforms.day(Types.TimestampType.withZone()));
+    return new UnboundTransform<>(ref(name), (Transform<?, T>) Transforms.day());
   }
 
   @SuppressWarnings("unchecked")
   public static <T> UnboundTerm<T> hour(String name) {
-    return new UnboundTransform<>(
-        ref(name), (Transform<?, T>) Transforms.hour(Types.TimestampType.withZone()));
+    return new UnboundTransform<>(ref(name), (Transform<?, T>) Transforms.hour());
   }
 
   public static <T> UnboundTerm<T> truncate(String name, int width) {
-    return new UnboundTransform<>(ref(name), Transforms.truncate(Types.LongType.get(), width));
+    return new UnboundTransform<>(ref(name), Transforms.truncate(width));
```

Comment:
This is one of the places where the old API was awkward. Expressions with transforms needed to guess the source type before the expression was bound.